src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -1485,6 +1529,170 @@ void hiopAlgFilterIPMQuasiNewton::outputIteration(int lsStatus, int lsNum, int u
   }
 }
 
+#ifdef HIOP_USE_AXOM
+
+bool hiopAlgFilterIPMQuasiNewton::save_state_to_file(const ::std::string& path) noexcept
+{
+  auto success = true;
+  sidre::DataStore* ds = nullptr;
+  try {
+    ds = new sidre::DataStore();
+    this->save_state_to_data_store(ds);
+
+    sidre::IOManager writer(this->get_nlp()->get_comm());
+    int n_files;
+    MPI_Comm_size(this->get_nlp()->get_comm(), &n_files);
+    writer.write(ds->getRoot(), n_files, path.c_str(), sidre::Group::getDefaultIOProtocol());
+  } catch(const std::exception& exp) {
+    nlp->log->printf(hovError, "Error in saving checkpoint to file '%s'\n", path.c_str());
+    nlp->log->printf(hovError,  "  Addtl info: %s\n", exp.what());
+    success = false;    
+  }  
+  delete ds;
+  return success;
+}
+
+bool hiopAlgFilterIPMQuasiNewton::load_state_from_file(const ::std::string& path) noexcept
+{
+  sidre::DataStore* ds = nullptr;
+  auto success = true;
+  try {
+    ds = new sidre::DataStore();
+    sidre::IOManager reader(this->get_nlp()->get_comm());
+    reader.read(ds->getRoot(), path, false);
+    
+    this->load_state_from_data_store(ds);
+    
+  } catch(const std::exception& exp) {
+    nlp->log->printf(hovError, "Error in loading checkpoint from file '%s'\n", path.c_str());
+    nlp->log->printf(hovError,  "  Addtl info: %s\n", exp.what());
+    success = false;
+  }
+  delete ds;
+  return success;
+}
+
+void hiopAlgFilterIPMQuasiNewton::
+copy_vec_to_new_view(const ::std::string& name, const hiopVector* vec, sidre::Group* nlp_group)
+{
+  using IndType = sidre::IndexType;
+  const IndType size = vec->get_local_size();
+  sidre::View* dest = nlp_group->createViewAndAllocate(name, sidre::DOUBLE_ID, size);
```

Comment:
In LiDO, we often use the same DataStore for the whole run. If we were to call this multiple times, I think it will fail because the views would already exist.
Right now this method always makes a new view. It would be better if it checked first if the view already exists. If it already exists in the Group, use that one. If it doesn't already exist, create a new view.

Reply:
last commit does that. note that an exception will be thrown if the view exists and has a different number of elements than HiOp's state variable.